xbmc/GUIInfoManager.h

```diff
@@ -152,6 +152,7 @@ class CGUIInfoManager : public IMsgTargetCallback, public Observable,
   std::string GetDuration(TIME_FORMAT format = TIME_FORMAT_GUESS) const;
 
   void SetCurrentItem(CFileItem &item);
+  void SetCurrentItemInternal(CFileItem &item);
```

Comment:
Keeping it separate is the best way to go,  we have way too many methods that take Boolean parameters which requires docs and or looking at implementation to know what it does. Make it obvious when seeing the invocation that it runs off thread, like async, background, nonblocking, just pick something nice